python/pyarrow/acero.py

```diff
@@ -374,7 +374,7 @@ def _filter_table(table, expression):
         Declaration("filter", options=FilterNodeOptions(expression))
     ])
     result = decl.to_table(use_threads=True)
-    if is_batch:
+    if is_batch and result.num_rows > 0:
```

Comment:
With this check we will return an empty `Table` instead of an empty `RecordBatch`. We might use try and catch or add an additional else so that in case the the input is a `RecordBatch` and the filtering result is empty, we still return an empty `RecordBatch`.

Reply:
@AlenkaF I might have miss-understood here but shouldn't this return a Table on that case (as per docstring?). Otherwise we should update the docstring to specify that we can return a RecordBatch.
The initial commit returned a RecordBatch on that case, see:
https://github.com/apache/arrow/pull/46057/commits/4472ac84067e9fd33061bca2ea7cb143afdf86d4

Reply:
Ah I should have updated the docstring. The original implementation (without my change) returns a RecordBatch when the input is a RecordBatch.

Reply:
> Will look into the C++ docs.

Found nothing that would help us here.